snakemake/workflow.py

```diff
@@ -1209,7 +1209,7 @@ def execute(
                 and self.remote_execution_settings.job_deploy_sources
                 and not executor_plugin.common_settings.can_transfer_local_files
             )
-            if should_deploy_sources:
+            if should_deploy_sources and not self.dryrun:
```

Comment:
Why can't we do something like this?
```suggestion
            if should_deploy_sources and not self.dryrun and len(self.dag):
```

Reply:
Then we won't even have to deploy the sources if the dag does not contain any needrun jobs.